### PR TITLE
feat: support client interceptors

### PR DESCRIPTION
**Pull request by zbigniew-mandziejewicz-3r**
Support client-side unary and streaming interceptors with similar interface to JS grpc-web interface.
Interceptors are executed by `Client.$createStreamingCall` and `Client.$createUnaryCall`.

Backwards compatibility is retained by keeping `Client.$createCall` for code generated with incompatible `protoc-gen-dart`.

### CODE REVIEW

**Comment by zbigniew-mandziejewicz-3r** [comment]
Corresponding PR for grpc/grpc-dart (https://github.com/grpc/grpc-dart/pull/338)

**Comment by zbigniew-mandziejewicz-3r** [comment]
How can i regenerate golden files?

**Comment by zbigniew-mandziejewicz-3r** [comment]
@sigurdm can I get feedback on the PR please?

**Comment by zbigniew-mandziejewicz-3r** [comment]
@kevmoo this PR has been dormant for 9 days, could you please take a look?

**Comment by kevmoo** [comment]
CC @mit-mit

File: protoc_plugin/lib/grpc_generator.dart
Comment (mraleph) [inline protoc_plugin/lib/grpc_generator.dart:238]: You can use `r'...'` then you don't need to escape `$`

**Review by mraleph — APPROVED** [review]
LGTM, we have to wait on landing this until we land gRPC changes.

File: protoc_plugin/lib/grpc_generator.dart
Comment (zbigniew-mandziejewicz-3r) [reply protoc_plugin/lib/grpc_generator.dart]: Made explicit codegen for each case (unary/streaming), as it was quite unreadable.

**Comment by mraleph** [comment]
I am going to land this once I land and release gRPC version with API changes.

**Comment by shaxbee** [comment]
@googlebot I signed it!

**Comment by shaxbee** [comment]
@mraleph Need to close this PR and recreate it as email address changed and it's tripping CLA bot.

**Comment by mraleph** [comment]
You could also amend you commits and force push (Though I am not sure it would work with `master` branch on github - it might be protected).

**Comment by shaxbee** [comment]
@mraleph I did that but original author of PR doesn't match commit author and it's still tripping the CLA bot.

**Comment by shaxbee** [comment]
@googlebot I signed it!

**Comment by shaxbee** [comment]
@googlebot I signed it!

**Comment by mraleph** [comment]
@shaxbee I have looked at the internal dashboard. There are two problems:

- It seems the bot can't figure out @zbigniew-mandziejewicz-3r's email address and confirm CLA on file. 
- Your new user is not a PR author so you need to tell the bot that you consent to your changes being included into this PR. (Try saying `@googlebot I consent!`).

Alternatively just nuke this PR and create a new one.

**Pull request closed without merging**
